src/hotspot/share/runtime/os.cpp

```diff
@@ -1016,12 +1016,19 @@ static void print_hex_location(outputStream* st, const_address p, int unitsize,
 }
 
 void os::print_hex_dump(outputStream* st, const_address start, const_address end, int unitsize,
-                        bool print_ascii, int bytes_per_line, const_address logical_start) {
+                        bool print_ascii, int bytes_per_line, const_address logical_start,  const_address highlight_address) {
   constexpr int max_bytes_per_line = 64;
   assert(unitsize == 1 || unitsize == 2 || unitsize == 4 || unitsize == 8, "just checking");
   assert(bytes_per_line > 0 && bytes_per_line <= max_bytes_per_line &&
          is_power_of_2(bytes_per_line), "invalid bytes_per_line");
 
+  // highlight_address should be within the printed range
+  if (highlight_address != nullptr) {
+    if (highlight_address < start || highlight_address > end) {
```

Comment:
Range is `[)`, so it should be `highlight_address >= end`

But I would just assert in debug builds, not do anything in release builds:

Its an error - the caller is responsible for giving us a highlight address that is in range. So in debug builds we get an error. But the error is benign, since all it causes is the function ignoring the highlight hint - since it never encounters the address. Therefore, setting it to NULL is superfluous.